docs/products/access/topics/custom-registration/custom-registration-v2.md

```diff
@@ -0,0 +1,145 @@
+# Custom Registration v1
+
+## Authentication
+
+This endpoint requires the client to provide the Client Authentication in the form of a PrivateKeyJWT 
+(`urn:ietf:params:oauth:client-assertion-type:jwt-bearer`) assertion.
+
+## Init Step
+The Init Step is only used for the `TWO_STEP` flow. The `ONE_STEP` flow uses the Complete Step.
+
+Endpoint: `POST /oauth/v2/custom-registration/{idp}/init`
+
+| Parameter | Description                  |
+|-----------|------------------------------|
+| `idp`     | Identity provider identifier |
+
+!!! info
+
+JSON body parameters:
+
+| Param                   | Required | Description                                                                                                                                                  |
+|-------------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `client_assertion_type` | yes      | The type of the client assertion that is part of the request, currently the only supported value is `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` |
+| `client_assertion`      | yes      | Client assertion in the form of a JWT signed by the client's private key                                                                                     |
+| `data`                  | no       | Raw registration request data which will be provided to the Extension Engine                                                                                 |
+
+Example request:
+```http
+POST /oauth/v2/custom-registration/example-custom-registration-idp/init HTTP/1.1
+Host: onegini.example.com
+Content-Type: application/json
+
+{
+   "client_assertion_type" : "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+   "client_assertion" : "eyJhbGciOiJSUzI1NiIsImtpZCI6IjIyIn0.
+     eyJpc3Mi[...omitted for brevity...].
+     cC4hiUPo[...omitted for brevity...]",
+  "data" : "{\"custom_json_key\":\"custom json data\"}" // optional
+ }
+```
+
+Example success response:
+```http
+HTTP/1.1 200 Ok
+Content-Type: application/json;charset=UTF-8
+Cache-Control: no-store
+Pragma: no-cache
+
+{
+   "transaction_id": "123123"   //something unique, should be passed with complete step
```

Comment:
That's not proper JSON. It's missing the commas after each key-value pair. The comments aren't really necessary because we also explain each property in the table below.